boards/nano33ble/src/main.rs

```diff
@@ -0,0 +1,283 @@
+//! Tock kernel for the Arduino Nano 33 BLE.
+//!
+//! It is based on nRF52840 SoC (Cortex M4 core with a BLE transceiver).
+
+#![no_std]
+#![no_main]
+#![deny(missing_docs)]
+
+use kernel::capabilities;
+use kernel::common::dynamic_deferred_call::{DynamicDeferredCall, DynamicDeferredCallClientState};
+use kernel::component::Component;
+use kernel::hil::gpio::ActivationMode::ActiveLow;
+#[allow(unused_imports)]
+use kernel::{create_capability, debug, debug_gpio, debug_verbose, static_init};
+
+use nrf52840::gpio::Pin;
+
+// Three-color LED.
+const LED_RED_PIN: Pin = Pin::P0_24;
+const LED_GREEN_PIN: Pin = Pin::P0_16;
+const LED_BLUE_PIN: Pin = Pin::P0_06;
+
+const LED_KERNEL_PIN: Pin = Pin::P0_13;
+
+// const BUTTON_RST_PIN: Pin = Pin::P0_18;
+
+const GPIO_D2: Pin = Pin::P1_11;
+const GPIO_D3: Pin = Pin::P1_12;
+const GPIO_D4: Pin = Pin::P1_15;
+const GPIO_D5: Pin = Pin::P1_13;
+const GPIO_D6: Pin = Pin::P1_14;
+const GPIO_D7: Pin = Pin::P0_23;
+const GPIO_D8: Pin = Pin::P0_21;
+const GPIO_D9: Pin = Pin::P0_27;
+const GPIO_D10: Pin = Pin::P1_02;
+
+const UART_TX_PIN: Pin = Pin::P1_03;
+const UART_RX_PIN: Pin = Pin::P1_10;
+
+/// UART Writer for panic!()s.
+pub mod io;
+
+// State for loading and holding applications.
+// How should the kernel respond when a process faults.
+const FAULT_RESPONSE: kernel::procs::FaultResponse = kernel::procs::FaultResponse::Panic;
+
+// Number of concurrent processes this platform supports.
+const NUM_PROCS: usize = 8;
+
+#[link_section = ".app_memory"]
+static mut APP_MEMORY: [u8; 245760] = [0; 245760];
+
+static mut PROCESSES: [Option<&'static dyn kernel::procs::ProcessType>; NUM_PROCS] = [None; 8];
+
+static mut CHIP: Option<&'static nrf52840::chip::Chip> = None;
+
+/// Dummy buffer that causes the linker to reserve enough space for the stack.
+#[no_mangle]
+#[link_section = ".stack_buffer"]
+pub static mut STACK_MEMORY: [u8; 0x1000] = [0; 0x1000];
+
+/// Supported drivers by the platform
+pub struct Platform {
+    // ble_radio: &'static capsules::ble_advertising_driver::BLE<
+    //     'static,
+    //     nrf52::ble_radio::Radio,
+    //     VirtualMuxAlarm<'static, Rtc<'static>>,
+    // >,
+    // ieee802154_radio: Option<&'static capsules::ieee802154::RadioDriver<'static>>,
```

Comment:
I realize this is commented out anyway, but no need for it to be an Option when you aren't using the `nrf52dk_base` setup, if you want to exclude it you will just comment it out.